src/Connections/FeedConnection.ts

```diff
@@ -0,0 +1,68 @@
+import {Appservice} from "matrix-bot-sdk";
+import { IConnection, IConnectionState } from ".";
+import { BridgeConfigFeeds } from "../Config/Config";
+import { FeedEntry, FeedError} from "../feeds/FeedReader";
+import LogWrapper from "../LogWrapper";
+import { GetConnectionsResponseItem } from "../provisioning/api";
+import { IBridgeStorageProvider } from "../Stores/StorageProvider";
+import { BaseConnection } from "./BaseConnection";
+import markdown from "markdown-it";
+
+const log = new LogWrapper("FeedConnection");
+const md = new markdown();
+
+export interface FeedConnectionState extends IConnectionState {
+    url: string;
+}
+
+export class FeedConnection extends BaseConnection implements IConnection {
+    static readonly CanonicalEventType = "uk.half-shot.matrix-hookshot.feed";
+    static readonly EventTypes = [ FeedConnection.CanonicalEventType ];
+    private hasError = false;
+
+    public get feedUrl(): string {
+        return this.state.url;
```

Comment:
I would put some linting code around here to trim out any crap from a user provided URL (e.g. `#` values), converting to lowercase, trimming etc. Hopefully this means we can dedupe URLs further on.

Reply:
I've done it in FeedReader in a62235e925d810ce39059784ff7db7594517b22e, while `Set`ing them to ensure that we're not fetching a feed twice if it appears in two rooms. I stripped HTML anchors but left the casing intact, as `/feed.xml?secretToken=aaa` and /feed.xml?secretToken=AAA` are in fact two different URLs and potentially different feeds.